src/registrar/models/domain_application.py

```diff
@@ -572,6 +572,19 @@ def domain_is_not_active(self):
             return not self.approved_domain.is_active()
         return True
 
+    def delete_and_clean_up_domain(self, called_from):
+        try:
+            domain_state = self.approved_domain.state
+            # Only reject if it exists on EPP
+            if domain_state != Domain.State.UNKNOWN:
+                self.approved_domain.deletedInEpp()
+                self.approved_domain.save()
```

Comment:
(Q) I don't think this save is needed because the domain itself is getting deleted, right?

Reply:
Negatory. That method now also sets a deleted date that we store in the DB.